extensions/ql-vscode/src/model-editor/extensions-workspace-folder.ts

```diff
@@ -67,6 +61,54 @@ async function getRootWorkspaceDirectory(): Promise<Uri | undefined> {
   // If we are at the root of the filesystem, we can't go up any further and there's something
   // wrong, so just return undefined
   if (commonRootUri.fsPath === Uri.joinPath(commonRootUri, "..").fsPath) {
+    return undefined;
+  }
+
+  return commonRootUri;
+}
+
+/**
+ * Finds the root directory of this workspace. It is determined
+ * heuristically based on the on-disk workspace folders.
+ *
+ * The heuristic is as follows:
+ * 1. If there is a workspace file (`<basename>.code-workspace`), use the directory containing that file
+ * 1. If there is only 1 workspace folder, use that folder=
+ * 3. If there is a common root directory for all workspace folders, use that directory
+ *   - Workspace folders in the system temp directory are ignored
+ *   - If the common root directory is the root of the filesystem, then it's not used
+ * 5. If there is a .git directory in any workspace folder, use the directory containing that .git directory
+ *    for which the .git directory is closest to a workspace folder
+ * 6. If none of the above apply, return `undefined`
```

Comment:
minor: the numbering looks a bit off here 🔢 😄

Reply:
Thanks, I've changed this to be correctly numbered.